### PR TITLE
Upgrade Node engine to 14

### PR DESCRIPTION
**Pull request by jessethach**
## Background
- Temporally point to upgraded component library (will need to change once lib is deployed properly with a Major increment change)
- Fix type errors for buttons
- Edit nvmrc
- Edit config yml

## GitHub Issue
#1458 

## Associated PRs
Will be ready to merge after component library is upgraded https://github.com/ctoec/component-library/pull/178
- https://github.com/skylight-hq/ctoec-devops/pull/224

## Validation Plan

- [ ] Make sure logging in still works on all user levels
- [ ] **Admin**: Test reading and modifying organizations
- [ ] **Admin**: Test reading and modifying users
- [ ] **Provider**: Test importing roster
- [ ] **Provider**: Test importing reading and modifying children in roster
- [ ] Keep eye on console for any unfamiliar activity (Unrecognized console errors for example)
- [ ] Keep eye on Network Tab for any failed requests

## Tech Specs
- [ ] **Optional**: Install NVM to test nvmrc. Run `nvm use` to use version targeted in rc file.

## Automated Testing
- [ ] All unit tests are passing.
- [ ] Unit tests have been added to cover this changeset.

## Additional Context
Currently, this pr points to the component library on a specific branch. Once the component library is validated and merged, this PR will be ready to be reviewed and validated.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -2,8 +2,11 @@
   "name": "client",
   "version": "0.1.0",
   "private": true,
+  "engines": {
+    "node": ">=14.18.1 <16.0.0"
+  },
   "dependencies": {
-    "@ctoec/component-library": "^1.1.29",
+    "@ctoec/component-library": "https://github.com/ctoec/component-library.git#jesse/1458-upgrade-node-14",
```
Comment (jessethach) [inline client/package.json:9]: Will need to change this line to point at new major version of the component lib once this pr is merged https://github.com/ctoec/component-library/pull/178

**Review by gordonfarrell — APPROVED** [review]
All these changes look good to me! 🎉 

The only thing I'm seeing to keep in mind is that when we do merge and update the component library version number we want to make sure its pointing to the right one. I had thought base was already set to use component library 1.31 which I believe is the most current, but maybe not? So after your changes we should presumably be at 1.32. You're probably already tracking this but I wanted to make sure. Cheers!

**Review by lukenewman — APPROVED** [review]

**Comment by gordonfarrell** [comment]
## Validation Plan

- [ ] Make sure logging in still works on all user levels
- [ ] **Admin**: Test reading and modifying organizations
- [ ] **Admin**: Test reading and modifying users
- [ ] **Provider**: Test importing roster
- [ ] **Provider**: Test importing reading and modifying children in roster
- [ ] Keep eye on console for any unfamiliar activity (Unrecognized console errors for example)
- [ ] Keep eye on Network Tab for any failed requests

**Comment by gordonfarrell** [comment]
Everything seems to be operating as normal with this upgrade 🦈🦈🦈🦈🦈

**Pull request merged**